sim/sim.py

```diff
@@ -311,14 +327,25 @@ def power_in(self):
         return self.controller.power[self.myT]
     
      # ---
-    def data_rate(self, conditions = []):
+    def data_rate(self,time_index,conditions=[]):
         """ Calculate the total data rate, traversing over the device collection. """
         dr = 0.0
         for dk in self.devices.keys():
             if dk=='UT' and 'TX' in conditions:
-                dr+=self.devices[dk].data_rate_tx()
+                #dr+=self.devices[dk].data_rate_tx() # not sure we need this?
```

Comment:
Ok, now data_rate_tx()  is deprecated -- just get rid of it in the in the devices.py